### PR TITLE
To do list

### PR DESCRIPTION
**Pull request by kundziorro**

### CODE REVIEW

File: to_do_list/app.py [hunk 1]
```diff
@@ -1,70 +1,94 @@
...
-    try:
-        choose_task = input("Choose task to update. ")
-        tasks.pop(choose_task)
-        new_status = input("Set new status. ")
-        tasks.update({choose_task: new_status})
-    except KeyError:
-        print("Task not found. ")
+    def update_task(
+        self,
+    ):
```
Comment (jacekplocharczyk) [inline to_do_list/app.py:29]: this method constructor should be in one line

File: to_do_list/app.py [hunk 2]
```diff
@@ -1,70 +1,94 @@
...
-    file = open("To do list.txt", "w+")
-    for task, status in tasks.items():
-        file.write(task + "-" + status + "\n")
-    file.close()
+    def show_all_tasks(self):
+        if not self.tasks:  # 4. simplfy condition in ifs
+            print("List is empty. \n")
+        else:
+            for task, status in self.tasks.items():
+                print(task + " - " + status)
```
Comment (jacekplocharczyk) [inline to_do_list/app.py:25]: you can use f-string here
see three approaches to strings - https://realpython.com/python-f-strings/

File: to_do_list/app.py [hunk 3]
```diff
@@ -1,70 +1,94 @@
...
-        print("Task not found. ")
+    def update_task(
+        self,
+    ):
+        self.show_all_tasks()
+        try:
+            choose_task = input("Choose task to update. ")
+            self.tasks.get(
+                choose_task
+            )  # 6. Use key to get to the item in dict instead of `pop`  in update_task()
```
Comment (jacekplocharczyk) [inline to_do_list/app.py:35]: Method `get` in dictionaries returns you a value.
You don't use it so there is probably a better way to check if a key is in dictionary:

`if key in dict:` - this way you will get rid of `try/except` and not using `get` returned value

File: to_do_list/app.py [hunk 4]
```diff
@@ -1,70 +1,94 @@
...
-def remove_task():
-    show_all_tasks()
-    try:
-        choose_task = input("Choose task to remove. ")
-        tasks.pop(choose_task)
-    except KeyError:
-        print("Task not found. ")
+    def remove_task(self):
+        self.show_all_tasks()
+        try:
```
Comment (jacekplocharczyk) [inline to_do_list/app.py:44]: The same as above.
Try using `if key in dict:` instead

File: to_do_list/app.py [hunk 5]
```diff
@@ -1,70 +1,94 @@
...
+            """
+            )
 
-#psikus
+            user_choice = int(input("\n Set proper option. \n"))
 
+            if user_choice == 1:
+                task_name = input("Set task name: ")
+                task_status = input("Set task status: ")
+                ToDoList.add_new_task(self, task_name, task_status)
```
Comment (jacekplocharczyk) [inline to_do_list/app.py:77]: Why are you using `ToDoList` here instead of `self`?!

File: to_do_list/app.py [hunk 6]
```diff
@@ -1,70 +1,94 @@
...
+        try:
+            choose_task = input("Choose task to update. ")
+            self.tasks.get(
+                choose_task
+            )  # 6. Use key to get to the item in dict instead of `pop`  in update_task()
+            new_status = input("Set new status. ")
+            self.tasks.update({choose_task: new_status})
+            print("Task updated. \n")
+        except KeyError:
+            print("Task not found. \n")
```
Comment (jacekplocharczyk) [inline to_do_list/app.py:40]: space before `\n` is redundant

File: to_do_list/app.py (hunk 3 above)
Comment (jacekplocharczyk) [reply to_do_list/app.py:35]: hint: `if choosen_task in self.tasks:`

File: to_do_list/app.py (hunk 1 above)
Comment (kundziorro) [reply to_do_list/app.py]: after black apply, due to `self,` instead of`self`

File: to_do_list/app.py (hunk 3 above)
Comment (kundziorro) [reply to_do_list/app.py]: done

File: to_do_list/app.py (hunk 4 above)
Comment (kundziorro) [reply to_do_list/app.py]: done

File: to_do_list/app.py (hunk 5 above)
Comment (kundziorro) [reply to_do_list/app.py]: done

File: to_do_list/app.py (hunk 6 above)
Comment (kundziorro) [reply to_do_list/app.py]: fixed

**Pull request merged**